pkg/syncer/spec/spec_controller.go

```diff
@@ -157,6 +151,22 @@ func NewSpecSyncer(gvrs []schema.GroupVersionResource, workloadClusterLogicalClu
 		klog.V(2).InfoS("Set up downstream informer", "clusterName", workloadClusterLogicalClusterName, "pcluster", workloadClusterName, "gvr", gvr.String())
 	}
 
+	secretMutator := specmutators.NewSecretMutator()
+
+	secretIndexer := upstreamInformers.ForResource(schema.GroupVersionResource{Group: "", Version: "v1", Resource: "secrets"}).Informer().GetIndexer()
+	deploymentMutator := specmutators.NewDeploymentMutator(upstreamURL, secretIndexer, workloadClusterLogicalClusterName, byWorkspace)
```

Comment:
why pass the indexer? Just make the index func a convention in the syncer. We will get scoping anyway soon and it will go away.

Reply:
workloadClusterLogicalClusterName seems to be wrong. This is about user secrets, not about the WorkloadCluster.